### ISSUE TITLE
Cannot achieve reported loss (<0.006) and performance on LIBERO with the finetuned model by given script.

### ISSUE THREAD

**Issue by Liang-ZX** [moojink/openvla-oft#74]
Hi,

Thanks for your excellent work. We have used the script to finetune the openvla/openvla-7b model on LIBERO-spatial,

```shell
torchrun --standalone --nnodes 1 --nproc-per-node X vla-scripts/finetune.py \
  --vla_path openvla/openvla-7b \
  --data_root_dir /PATH/TO/RLDS/DATASETS/DIR/ \
  --dataset_name libero_spatial_no_noops \
  --run_root_dir /YOUR/CHECKPOINTS/AND/LOG/DIR/ \
  --use_l1_regression True \
  --use_diffusion False \
  --use_film False \
  --num_images_in_input 2 \
  --use_proprio True \
  --batch_size 8 \
  --learning_rate 5e-4 \
  --num_steps_before_decay 100000 \
  --max_steps 150005 \
  --save_freq 10000 \
  --save_latest_checkpoint_only False \
  --image_aug True \
  --lora_rank 32 \
  --wandb_entity "YOUR_WANDB_ENTITY" \
  --wandb_project "YOUR_WANDB_PROJECT" \
  --run_id_note parallel_dec--8_acts_chunk--continuous_acts--L1_regression--3rd_person_img--wrist_img--proprio_state
```

 But after 150K step, our loss is still 0.008 and cannot achieve <0.006.
With this checkpoint, we evaluate our model on LIBERO-spatial and only ahieve ~92.6% success rate which is quite lower than the reported 96.2%. Could you please help to check what the problem is? Thank you.

![Image](https://github.com/user-attachments/assets/e93e8d4b-16ff-4b7d-abac-17c7d5a2e7c9)

The only difference may be that we use 8 A800 to finetune the model. But is this the problem?

**Comment by ZefuLin** [comment]
maybe you can try to begin with learning_rate=5e-5

**Comment by humphreyde** [comment]
me too, and I got just 0.08.

**Comment by dreamcubeblock** [comment]
It is interesting that we even surpassed the result in the paper and achieved 98.8% on libero_spatial:

![Image](https://github.com/user-attachments/assets/543cfb7a-42fe-4a2d-a96d-65684d63227b)

with single A100 GPU!
```
torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
  --vla_path openvla/openvla-7b \
  --data_root_dir /root/openvla/modified_libero_rlds \
  --dataset_name libero_spatial_no_noops \
  --run_root_dir /mnt/data/openvla-oft/checkpoints \
  --use_l1_regression True \
  --use_diffusion False \
  --use_film False \
  --num_images_in_input 2 \
  --use_proprio True \
  --batch_size 8 \
  --learning_rate 5e-4 \
  --num_steps_before_decay 100000 \
  --max_steps 150005 \
  --save_freq 10000 \
  --save_latest_checkpoint_only False \
  --image_aug True \
  --lora_rank 32 \
  --wandb_entity yours \
  --wandb_project openvla-oft \
  --run_id_note parallel_dec--8_acts_chunk--continuous_acts--L1_regression--3rd_person_img--wrist_img--proprio_state
```
Although the loss was only 0.03

![Image](https://github.com/user-attachments/assets/65e8a9d4-2801-489c-8b0e-c3d5a6714fd7)

Fine-tuned on libero_spatial and evaluated on libero_spatial.

There is an argument that single GPU training does work better

OH we set the shuffle buffer size to 1000, maybe thats the problem?

**Comment by LukeLIN-web** [comment]
> It is interesting that we even surpassed the result in the paper and achieved 98.8% on libero_spatial:
> 
> ![Image](https://github.com/user-attachments/assets/543cfb7a-42fe-4a2d-a96d-65684d63227b)
> 
> with single A100 GPU!
> 
> ```
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
>   --vla_path openvla/openvla-7b \
>   --data_root_dir /root/openvla/modified_libero_rlds \
>   --dataset_name libero_spatial_no_noops \
>   --run_root_dir /mnt/data/openvla-oft/checkpoints \
>   --use_l1_regression True \
>   --use_diffusion False \
>   --use_film False \
>   --num_images_in_input 2 \
>   --use_proprio True \
>   --batch_size 8 \
>   --learning_rate 5e-4 \
>   --num_steps_before_decay 100000 \
>   --max_steps 150005 \
>   --save_freq 10000 \
>   --save_latest_checkpoint_only False \
>   --image_aug True \
>   --lora_rank 32 \
>   --wandb_entity yours \
>   --wandb_project openvla-oft \
>   --run_id_note parallel_dec--8_acts_chunk--continuous_acts--L1_regression--3rd_person_img--wrist_img--proprio_state
> ```
> 
> Although the loss was only 0.03
> 
> ![Image](https://github.com/user-attachments/assets/65e8a9d4-2801-489c-8b0e-c3d5a6714fd7)
> 
> Fine-tuned on libero_spatial and evaluated on libero_spatial.
> 
> There is an argument that single GPU training does work better
> 
> OH we set the shuffle buffer size to 1000, maybe thats the problem?

From this, I think the reason is larger batch size will make model stick to a sharp suboptimal point.  
Because libero dataset is pretty small.  The model is overfitting

**Comment by Liang-ZX** [comment]
Thanks all. I have retrained the model and achieved 97.6% success on 140000 steps.

**Comment by LukeLIN-web** [comment]
> Thanks all. I have retrained the model and achieved 97.6% success on 140000 steps.

What is the difference in this success training?

**Comment by moojink** [comment]
Hi @Liang-ZX, glad you were able to get better results!

@dreamcubeblock Great to hear! Thank you for reporting that!

For others: There may be some variance between training runs and even different GPUs. Sometimes it helps to use a larger global batch size (either through more GPUs or gradient accumulation). Or sometimes it helps to train with the base learning rate longer, and then decay at a later time -- depends on the observed training dynamics!

**Comment by haofuly** [comment]
> It is interesting that we even surpassed the result in the paper and achieved 98.8% on libero_spatial:
> 
> ![Image](https://github.com/user-attachments/assets/543cfb7a-42fe-4a2d-a96d-65684d63227b)
> 
> with single A100 GPU!
> 
> ```
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
>   --vla_path openvla/openvla-7b \
>   --data_root_dir /root/openvla/modified_libero_rlds \
>   --dataset_name libero_spatial_no_noops \
>   --run_root_dir /mnt/data/openvla-oft/checkpoints \
>   --use_l1_regression True \
>   --use_diffusion False \
>   --use_film False \
>   --num_images_in_input 2 \
>   --use_proprio True \
>   --batch_size 8 \
>   --learning_rate 5e-4 \
>   --num_steps_before_decay 100000 \
>   --max_steps 150005 \
>   --save_freq 10000 \
>   --save_latest_checkpoint_only False \
>   --image_aug True \
>   --lora_rank 32 \
>   --wandb_entity yours \
>   --wandb_project openvla-oft \
>   --run_id_note parallel_dec--8_acts_chunk--continuous_acts--L1_regression--3rd_person_img--wrist_img--proprio_state
> ```
> 
> Although the loss was only 0.03
> 
> ![Image](https://github.com/user-attachments/assets/65e8a9d4-2801-489c-8b0e-c3d5a6714fd7)
> 
> Fine-tuned on libero_spatial and evaluated on libero_spatial.
> 
> There is an argument that single GPU training does work better
> 
> OH we set the shuffle buffer size to 1000, maybe thats the problem?

Hi @dreamcubeblock , I am still confused why single GPU outperforms 8×GPUs. In my opinion, single gpu corresponds to smaller global batch size, which means we should set smaller learning rate and large steps to achieve the same performance as 8×GPUs? Thanks for your further opinion!

**Comment by Linchenchen** [comment]
> It is interesting that we even surpassed the result in the paper and achieved 98.8% on libero_spatial:
> 
> ![Image](https://github.com/user-attachments/assets/543cfb7a-42fe-4a2d-a96d-65684d63227b)
> 
> with single A100 GPU!
> 
> ```
> torchrun --standalone --nnodes 1 --nproc-per-node 1 vla-scripts/finetune.py \
>   --vla_path openvla/openvla-7b \
>   --data_root_dir /root/openvla/modified_libero_rlds \
>   --dataset_name libero_spatial_no_noops \
>   --run_root_dir /mnt/data/openvla-oft/checkpoints \
>   --use_l1_regression True \
>   --use_diffusion False \
>   --use_film False \
>   --num_images_in_input 2 \
>   --use_proprio True \
>   --batch_size 8 \
>   --learning_rate 5e-4 \
>   --num_steps_before_decay 100000 \
>   --max_steps 150005 \
>   --save_freq 10000 \
>   --save_latest_checkpoint_only False \
>   --image_aug True \
>   --lora_rank 32 \
>   --wandb_entity yours \
>   --wandb_project openvla-oft \
>   --run_id_note parallel_dec--8_acts_chunk--continuous_acts--L1_regression--3rd_person_img--wrist_img--proprio_state
> ```
> 
> Although the loss was only 0.03
> 
> ![Image](https://github.com/user-attachments/assets/65e8a9d4-2801-489c-8b0e-c3d5a6714fd7)
> 
> Fine-tuned on libero_spatial and evaluated on libero_spatial.
> 
> There is an argument that single GPU training does work better
> 
> OH we set the shuffle buffer size to 1000, maybe thats the problem?

Hi, I wonder how long it takes to train on single GPU. Thank you!
